File: VB-Examples/11_Formatting/SetBorderToDataBar.vb
Imports Spire.Xls
Imports Spire.Xls.Core
Imports Spire.Xls.Core.Spreadsheet.Collections

Namespace SetBorderToDataBar
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub
		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new instance of Workbook
            Dim workbook As New Workbook()
            ' Load an existing Excel file
            workbook.LoadFromFile("..\..\..\..\..\..\Data\Template_Xls_9.xlsx")

            ' Get the first worksheet in the workbook
            Dim sheet As Worksheet = workbook.Worksheets(0)

            ' Get the conditional formats collection for the worksheet
            Dim xcfs As XlsConditionalFormats = sheet.ConditionalFormats(0)
            ' Get the first conditional format from the collection
            Dim cf As IConditionalFormat = xcfs(0)
            ' Get the DataBar object from the conditional format
            Dim dataBar1 As Spire.Xls.DataBar = cf.DataBar
            ' Set the type of border for the data bar (solid)
            dataBar1.BarBorder.Type = Spire.Xls.Core.Spreadsheet.ConditionalFormatting.DataBarBorderType.DataBarBorderSolid
            ' Set the color of the border to red
            dataBar1.BarBorder.Color = Color.Red

            ' Set the number value of cell E1 to 200
            sheet("E1").NumberValue = 200

            ' Add a new conditional formats collection to the worksheet
            Dim xcfs2 As XlsConditionalFormats = sheet.ConditionalFormats.Add()
            ' Add the range E1 to the conditional formats collection
            xcfs2.AddRange(sheet.Range("E1"))
            ' Add a new conditional format to the collection
            Dim cf2 As IConditionalFormat = xcfs2.AddCondition()
            ' Set the format type to DataBar
            cf2.FormatType = ConditionalFormatType.DataBar
            ' Set the type of border for the data bar (solid)
            cf2.DataBar.BarBorder.Type = Spire.Xls.Core.Spreadsheet.ConditionalFormatting.DataBarBorderType.DataBarBorderSolid
            ' Set the color of the border to red
            cf2.DataBar.BarBorder.Color = Color.Red
            ' Set the color of the data bar to green-yellow
            cf2.DataBar.BarColor = Color.GreenYellow

            ' Save the workbook to a new file with the name "SetBorderToDataBar_result.xlsx"
            Dim result As String = "SetBorderToDataBar_result.xlsx"
            workbook.SaveToFile(result, ExcelVersion.Version2013)

            ' Release the resources used by the workbook
            workbook.Dispose()

            'View the document
            FileViewer(result)
		End Sub

		Private Sub FileViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class
End Namespace

File: CS-Examples/11_Formatting/SetBorderToDataBar.cs
using Spire.Xls;
using Spire.Xls.Core;
using Spire.Xls.Core.Spreadsheet.Collections;
using System;
using System.Drawing;
using System.Windows.Forms;

namespace SetBorderToDataBar
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }
        private void btnRun_Click(object sender, EventArgs e)
        {
            // Create a new workbook object
            Workbook workbook = new Workbook();

            // Load an existing Excel document
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\Template_Xls_9.xlsx");

            // Get the first sheet from the workbook
            Worksheet sheet = workbook.Worksheets[0];

            // Get the data bar format from the first conditional format
            XlsConditionalFormats xcfs = sheet.ConditionalFormats[0];
            IConditionalFormat cf = xcfs[0];
            Spire.Xls.DataBar dataBar1 = cf.DataBar;

            // Set the border type and color for the data bar format
            dataBar1.BarBorder.Type = Spire.Xls.Core.Spreadsheet.ConditionalFormatting.DataBarBorderType.DataBarBorderSolid;
            dataBar1.BarBorder.Color = Color.Red;

            // Set a new data bar format to cell E1
            sheet["E1"].NumberValue = 200;
            XlsConditionalFormats xcfs2 = sheet.ConditionalFormats.Add();
            xcfs2.AddRange(sheet.Range["E1"]);
            IConditionalFormat cf2 = xcfs2.AddCondition();
            cf2.FormatType = ConditionalFormatType.DataBar;
            cf2.DataBar.BarBorder.Type = Spire.Xls.Core.Spreadsheet.ConditionalFormatting.DataBarBorderType.DataBarBorderSolid;
            cf2.DataBar.BarBorder.Color = Color.Red;
            cf2.DataBar.BarColor = Color.GreenYellow;

            // Save the modified workbook to a file
            String result = "SetBorderToDataBar_result.xlsx";
            workbook.SaveToFile(result, ExcelVersion.Version2013);

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            //View the document
            FileViewer(result);
        }

        private void FileViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
    }
}
